### PR TITLE
fix: ensure stats collector doesn't aggregate stats from multiple runs

### LINKED ISSUE #749: Ensure stats collector doesn't aggregate stats from multiple runs
Follow up on https://github.com/Azure/aad-pod-identity/pull/744#pullrequestreview-470646883

### PR DESCRIPTION
**Pull request by chewong**
Signed-off-by: Ernest Wong <chuwon@microsoft.com>

**Reason for Change**:

Ensure stats collector doesn't aggregate stats from multiple runs.

**Requirements**

- [x] squashed commits
- [x] included documentation
- [x] added unit tests and e2e tests (if applicable). See [test standard](https://github.com/Azure/aad-pod-identity/blob/master/CONTRIBUTING.md#test-standard) for more details.
- [x] ran `make precommit`

**Issue Fixed**:

fixes #749

**Notes for Reviewers**:

### CODE REVIEW

File: pkg/stats/stats.go [hunk 1]
```diff
@@ -7,12 +7,21 @@ import (
...
+	// globalStats is a map that stores the duration of each statistic.
+	globalStats map[Type]time.Duration
 
-	// CountStats is a map that stores the count of each stats.
-	CountStats map[Type]int
+	// minMaxTimeStats is a map that stores the earliest start time and
+	// latest end time of statistics that are being collected concurrently.
+	minMaxTimeStats map[Type]minMaxTime
+
+	// CountStats is a map that stores the count of each statistic.
```
Comment (aramase) [inline pkg/stats/stats.go:23]: nit: s/CountStats/countStats

File: pkg/cloudprovider/vm.go [hunk 2]
```diff
@@ -113,8 +113,8 @@ func (c *VMClient) UpdateIdentities(rg, nodeName string, vm compute.VirtualMachi
 	if err = future.WaitForCompletionRef(ctx, c.client.Client); err != nil {
 		return fmt.Errorf("failed to wait for identity update completion for vm %s in resource group %s, error: %+v", nodeName, rg, err)
 	}
-	stats.UpdateCount(stats.TotalUpdateCalls, 1)
-	stats.Update(stats.CloudUpdate, time.Since(begin))
+	stats.Increment(stats.TotalPatchCalls, 1)
+	stats.Aggregate(stats.CloudPatch, time.Since(begin))
```
Comment (aramase) [inline pkg/cloudprovider/vm.go:117]: Wouldn't using `Aggregate` here keep adding the `CloudPATCH` duration for all the base concurrent calls? The final statistics we print will have a sum of all Cloud PATCHs from the concurrent update.

File: pkg/stats/stats.go [hunk 3]
```diff
@@ -36,149 +45,169 @@ const (
...
 
-// Init initializes the maps uesd to store the stats.
+// Init initializes the maps uesd to store the
 func Init() {
-	GlobalStats = make(map[Type]time.Duration)
-	CountStats = make(map[Type]int)
+	globalStats = make(map[Type]time.Duration)
+	minMaxTimeStats = make(map[Type]minMaxTime)
+	countStats = make(map[Type]int)
 	mutex = &sync.RWMutex{}
```
Comment (aramase) [inline pkg/stats/stats.go:105]: If we are not using read locks, we should just change this to `sync.Mutex`

File: pkg/cloudprovider/vm.go (hunk 2 above)
Comment (chewong) [reply pkg/cloudprovider/vm.go:117]: Right, let me change that.

File: pkg/stats/stats.go (hunk 3 above)
Comment (chewong) [reply pkg/stats/stats.go:105]: We used it in `Print()` method.

**Pull request merged**